storage/buffer_test.go

```diff
@@ -161,6 +161,43 @@ func TestBufferedSeriesIteratorNoBadAt(t *testing.T) {
 	it.Next()
 }
 
+func TestBufferReduceDelta(t *testing.T) {
```

Comment:
We test the reduceDelta indirectly via `TestBufferedSeriesIterator`. IMO, testing all the individual internal components is not a good idea when those already get tested via other exported methods unless the internal component is huge to deserve a test for itself.

Reply:
Thanks for your review, I'll close it.